app/controllers/map_controller.rb

```diff
@@ -16,7 +16,10 @@ def index
   end
 
   def show
-    @node = DrupalNode.find_map_by_slug(params[:name]+'/'+params[:date])
+    @node = DrupalNode.find params[:name]
```

Comment:
Yeah that's what I wanted to say that the history module will take care of old urls. The thing is the `find` method will work only for a valid friendly_id slug. Taking the first case where the date is appended with a '/' the find method will have to search something like `map-name/07-29-2016` which is not the a valid friendly_id slug in this case. The slug is just the parameterized version of the node title.

Not to contradict you Jeff, but now that it is clear to you that no old urls gets lost and get redirected to the new url is there any problem to use a map url like `/map/map-name--2/07-29-2016` for a repost of the map. I mean an identifier like `--2` beside the map name in the url won't be a big problem .